app/controllers/expenses_controller.rb

```diff
@@ -0,0 +1,81 @@
+class ExpensesController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_expense, only: %i[show edit update destroy]
+
+  # GET /expenses or /expenses.json
+  def index
+    @group = current_user.groups.find(params[:group_id])
+    @expenses = @group.expenses..order(created_at: :desc)
+  end
+
+  # GET /expenses/1 or /expenses/1.json
+  def show; end
+
+  # GET /expenses/new
+  def new
+    @group = current_user.groups.find_by(id: params[:group_id])
```

Comment:
- [ ] As mentioned above please refactor the `groups.find_by(id: params[:group_id])` piece of code when you find a group by group id which is used in almost every action of the controller. This makes your code much cleaner and robust